### ISSUE TITLE
Support Edit messages

### ISSUE THREAD

**Issue by knayan1** [zhukov/webogram#1155]
Hi Team,

Thanks for providing a unique feature of "editing" sent message in App version. Please check following suggestions in this:
1. Provide "Edit Sent message" option in **webogram**.
2. (App) **Restrict users to edit sent message only if its unread**. Here we can replace time limit of 2 days with message status. That is, If sent message has been read then what is the use of editing it? If we are providing then edit notification needs to be sent to user.
3. Provide "**Edit History**" for the edited message.

Thanks,
Kamal Nayan

**Comment by vinipsmaker** [comment]
> (App) Restrict users to edit sent message only if its unread. Here we can replace time limit of 2 days with message status. That is, If sent message has been read then what is the use of editing it? If we are providing then edit notification needs to be sent to user.

I disagree about this. I know it doesn't make sense, but people don't make sense (nor need to).

**Comment by knayan1** [comment]
**Consider a scenario**: User X sent any message to Y, that is not yet read by Y. X then edited it, now when Y came to read this, he saw that this is an edited message. Conversation went long. X now edited that message again, he may write any message that might contradicts his earlier message, might be something rude also. Y has no any such information about this edit, also if this message would have been critical, he would again scroll up to see that message. Again he saw that message is edited!

Telegram neither shown him the message edit history, nor notified him about the edit. But user is shocked to see the message on which he might have agreed upon (during long conversation), which he was not supposed to do.

I hope I am able to explain you its need.

Thanks,
Kamal Nayan

**Comment by Kondra007** [comment]
@zhukov So where's this feature? You put label "ready" but looks like I'm blind and don't see such feature in 0.5.5

**Comment by zhukov** [comment]
@Kondra007 'ready' means ready to implement. Otherwise the issue would be closed.

**Comment by knayan1** [comment]
Hey @Kondra007, I do agree with @zhukov, "ready" means ready to land / implement.

**Comment by mufid** [comment]
> Telegram neither shown him the message edit history, nor notified him about the edit. But user is shocked to see the message on which he might have agreed upon (during long conversation), which he was not supposed to do.

How about simply "Edited" tag before the timestamp? Just like we see on Facebook. (e.g. 10:50 [edited]). No need to notify user explicitly, they can see something was changed.

**Comment by Akronix** [comment]
I'd start at least with:
> 1. Provide "Edit Sent message" option in webogram.
and a minimal "Edited" tag along to the message to see that it was edited.

Later on, however, I'm up for the 3 proposal:
> 3. Provide "Edit History" for the edited message.
as Facebook does, If I remember well.

**Comment by knayan1** [comment]
From user point of view, providing "Edited" tag would not be sufficient. Yes, you can work on it in first phase, and providing "Edit history" in second phase, but the time gap between two releases should not be too long.

> No need to notify user explicitly, they can see something was changed.

They can see something changed, but can't see what is changed! So, "Edit history" will be solution to this issue.

**Comment by Akronix** [comment]
I was looking up [the telegram API](https://core.telegram.org/methods) but I didn't found any method to get if there's a message edited. Also note that last update of the API was in October, 2014 but editing messages was added in [May, 2016] (https://telegram.org/blog/edit)

I guess we still can find out how to know if there's a edited message looking to the other [Telegram's apps](https://telegram.org/apps) code. But I don't see any edit history in the rest of the client apps so I'd say that the server is not retrieving any edition history; and, if so, the only way to have a edition history is that the client itself (webogram) store such data, which makes things more complicated and, of course, you won't be able to see it as soon as you change to another browser or device.

**Comment by unmultimedio** [comment]
I was looking for the option to edit the messages in Webogram (I think that's the version thats embedded into [Franz](http://meetfranz.com/)) But I couldn't find it, so I came here to report it. Found this thread, and started searching in the API to see what was the method, so this can be easier to implement, but it seems to be that the API docs are not so updated.

Nevertheless, found this [question](http://stackoverflow.com/questions/38707199/telegram-api-edit-message) in StackOverflow, where somebody seemed to have found the API function name:

`messages.editMessage`

Hope this is helpful so this can be implemented.
